activiti-engine/src/main/java/org/activiti/engine/impl/cmd/SetProcessInstanceNameCmd.java

```diff
@@ -56,6 +58,10 @@ public Void execute(CommandContext commandContext) {
 
     // Actually set the name
     execution.setName(name);
+    
+    if (commandContext.getEventDispatcher().isEnabled()) {
```

Comment:
Here, the dispatchEvent call will fire entity updated event for the process instance execution entity inside execution context